offscreen_render_target/src/offscreen_render_target.mm

```diff
@@ -480,28 +447,30 @@ - (void)setFramebufferOnly:(BOOL)framebufferOnly
         return m_impl->get_oriented_image(orientation);
     }
 
-    void offscreen_render_target::init(){
-        m_impl->init();
-    }
 
-    void offscreen_render_target::activate_context(){
-        m_impl->activate_context();
+    void offscreen_render_target::init(int32_t width, int32_t height){
+        m_impl = std::make_unique<impl>();
+        m_impl->init(width, height);
     }
 
-    void offscreen_render_target::prepare_rendering(){
-        m_impl->prepare_rendering();
+    void offscreen_render_target::deinit() {
+        m_impl->deinit();
     }
 
-    void offscreen_render_target::orient_image(interfaces::orient_format orient){
-        m_impl->orient_image(orient);
+    void offscreen_render_target::surface_changed(int32_t width, int32_t height){
+        m_impl->surface_changed(width, height);
     }
 
-    void* offscreen_render_target::get_image(){
-        return m_impl->get_image();
-    }
+    void offscreen_render_target::activate_context()    {}
+    void offscreen_render_target::deactivate_context()  {}
+    void offscreen_render_target::prepare_rendering()   {}
+    
+    void offscreen_render_target::orient_image(bnb::oep::interfaces::rotation orient){}
+
+    pixel_buffer_sptr offscreen_render_target::read_current_buffer(bnb::oep::interfaces::image_format format){return nil;}
```

Comment:
It is necessary to shift the result from CVPixelBuffer to pixel_buffer in the required format